Katello 4.12 Release Notes

qcjames53 opened the pull request:
* [X] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

Added Katello 4.12 release notes and contributors.

jeremylenz commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,38 @@
 == Headline Features
 
-There are no highlights with Katello {KatelloVersion}.
+=== Content Access Modes Removed in Favor of Universal Simple Content Access
```
```suggestion
=== Entitlement Mode Removed in Favor of Simple Content Access
```

jeremylenz commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,38 @@
 == Upgrade Warnings
 
-There are no upgrade warnings with Katello {KatelloVersion}.
+Simple Content Access is now the only supported content access mode. Existing organizations may be automatically migrated on Katello upgrade (see more info above). Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations, and will never have any rows. It will be removed in a future release or release candidate. Instead, use the 'Host - Installed Prod ...
```
```suggestion
Simple Content Access is now the only supported content access mode. Existing organizations will be automatically migrated on Katello upgrade (see more info above). Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations, and will never have any rows. It will be removed in a future release or release candidate. Instead, use the 'Host - Installed Products' report. Additionally, please note that SCA-related API and Hammer endpoints and parameters have not yet been removed, so you will experience errors if you try to use them. They will be removed in a future release or release candidate.
```

jeremylenz reviewed (commented):
SCA-Only content is looking great, just a couple nits :)

qcjames53 commented:
@jeremylenz Rolled in your suggested changes, thanks. Also fixed a merge conflict.

qcjames53 commented on guides/doc-Release_Notes/master.adoc:
```diff
@@ -14,6 +14,7 @@ endif::[]
 
 // Start inserting specific x.y.z releases here
 include::topics/foreman-3.10.0.adoc[leveloffset=+1]
+include::topics/katello-4.12.0.adoc[leveloffset=+1]
```
I manually merged this PR with upstream when 3.10 was added so the order was reversed. Will fix.

qcjames53 commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,38 @@
 == Upgrade Warnings
 
-There are no upgrade warnings with Katello {KatelloVersion}.
+Simple Content Access is now the only supported content access mode. Existing organizations will be automatically migrated on Katello upgrade (see more info above). Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations, and will never have any rows. It will be removed in a future release or release candidate. Instead, use the 'Host - Installed Pro ...
```
I have separated the paragraphs into individual sentences, should be fixed. Thanks for the heads up.

qcjames53 commented:
@ekohl Thank you for all the rec's for the documentation on Katello 4.12. I have rebased the PR to include your suggestions. I split the contributors list into lines containing just people's names but if there should be another separator please let me know. That file has been blank for a release or two at this point.

qcjames53 commented:
@ekohl Did you want to take another look at this or are we ok to merge the changes in?

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
 
-There are no highlights with Katello {KatelloVersion}.
+=== Entitlement Mode Removed in Favor of Simple Content Access
+Simple Content Access is now the only content access mode supported.
```
```suggestion
Simple Content Access (SCA) is now the only content access mode supported.
```

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
+
+=== Debian Content Support Expanded
+Import and export are now supported for repositories containing debian-type content.
+Aditionally, the host details page will now display more details for debian packages.
```
```suggestion
Additionally, the host details page will now display more details for debian packages.
```

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
 == Upgrade Warnings
 
-There are no upgrade warnings with Katello {KatelloVersion}.
+Simple Content Access is now the only supported content access mode.
```
```suggestion
Simple Content Access (SCA) is now the only supported content access mode.
```

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
-There are no upgrade warnings with Katello {KatelloVersion}.
+Simple Content Access is now the only supported content access mode.
+Existing organizations will be automatically migrated on Katello upgrade (see more info above).
+Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations, and will never have any rows.
```
```suggestion
Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations and will never have any rows.
```

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
+Please note that the 'Subscription - Entitlements' report template does not work with SCA organizations, and will never have any rows.
+It will be removed in a future release or release candidate.
+Instead, use the 'Host - Installed Products' report.
+Additionally, please note that SCA-related API and Hammer endpoints and parameters have not yet been removed, so you will experience errors if you try to use them.
```
```suggestion
Additionally, please note that SCA-related API and Hammer endpoints and parameters have not yet been removed so you will experience errors if you try to use them.
```

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
-There are no deprecations with Katello {KatelloVersion}.
+Deprecated 'Subscription - Entitlements' report template as it does not work with SCA organizations.
+
+Deprecated certain SCA-related API and Hammer endpoints and parameters
```
```suggestion
Deprecated certain SCA-related API and Hammer endpoints and parameters.
```

qcjames53 commented:
@ekohl Did you want to take another look at the Katello 4.12 release documentation before we merge it in? We're releasing 4.12 today (hopefully) so we will likely need to merge this afternoon.

ekohl commented on guides/doc-Release_Notes/topics/katello-contributors.adoc:
I think you've listed every contributor ever, instead of just those who contributed in 4.12.

qcjames53 replied:
Thanks for taking a look at this. That is correct; I was unaware that this is only 4.12 contributors. I'll update this soon.

ekohl replied:
I hope you are aware of https://github.com/theforeman/theforeman.org/blob/gh-pages/scripts/committers.rb. I'd also like to discuss merging the contributor lists together into one list, but that's a separate discussion.

qcjames53 replied:
Thanks Ewoud. The contributors list should now be correct for 4.12 alone. Ok for a re-review.

That original big list of all contributors was generated from commands but it needed a decent bit of manual sleuthing since a few of the commits has incorrect names, nicknames, or emails as the full name field.

ekohl replied:
In https://github.com/theforeman/theforeman.org/blob/gh-pages/.mailmap I did make that mapping, but only for "recent" contributors. If you have more manual fixes you need to apply, that's the place to fix it.

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
 
-There are no deprecations with Katello {KatelloVersion}.
+Deprecated 'Subscription - Entitlements' report template as it does not work with SCA organizations.
+
```
```suggestion
```

ianballou replied:
I think we need the extra space there to logically separate the subs entitlement report template deprecation from the the next one:

![image](https://github.com/theforeman/foreman-documentation/assets/14796566/7359263e-a393-43da-97b0-3698837d4f75)

bangelic commented on guides/doc-Release_Notes/topics/katello.adoc:
```diff
@@ -4,14 +4,45 @@
+* SLA Status
+* Role Status
+* Usage Status
+* Addon Status
```
Was it spelled like this in the UI? Not spelled Add-on or Add on?

jeremylenz replied:
Unfortunately yes :confounded: 

![image](https://github.com/theforeman/foreman-documentation/assets/22042343/0230ce44-83fd-47f7-8b7f-df6020bad3af)

qcjames53 replied:
Addons status sounds icky to me. Just like fathers in law. I think it's ok as is?

jeremylenz replied:
https://www.theonion.com/william-safire-orders-two-whoppers-junior-1819565735

ianballou reviewed (approved):
Looking good to me!

ianballou commented:
Merging so we can be on time with GA since we have two acks -- if further improvements are requested we can include them in a separate PR.
